### PR TITLE
Fix UpdateUser Function

### LINKED ISSUE #579: Fix updateUser Function
Currently, the updateUser function is getting data to update in req.body, now in req.body you will get only profileDiffId so fetch the data from profile and update that.

### PR DESCRIPTION
**Pull request by RahulGoyal-tech**
closes #579

### CODE REVIEW

**Review by lakshayman — APPROVED** [review]
Looks good to me.

File: models/profileDiffs.js [hunk 1]
```diff
@@ -24,6 +24,22 @@ const fetchProfileDiffs = async () => {
...
 
+/**
+ * Fetches the profileDiff data of the provided profileDiff Id
+ * @param profileDiffId profileDiffId of the diffs need to be fetched
+ * @returns profileDiff Data
+ */
+const fetchProfileDiff = async (profileDiffId) => {
+  try {
+    const Diff = await profileDiffsModel.doc(profileDiffId).get();
+    const profileDiffData = Diff.data();
```
Comment (pallabez) [inline models/profileDiffs.js:35]:
```suggestion
    const profileDiff = await profileDiffsModel.doc(profileDiffId).get();
    const profileDiffData = profileDiff.data();
```

File: controllers/users.js [hunk 2]
```diff
@@ -196,8 +196,9 @@ const postUserPicture = async (req, res) => {
  */
 const updateUser = async (req, res) => {
   try {
-    const userId = req.params.userId;
-    const { id: profileDiffId, message, ...profileDiff } = req.body;
+    const { id: profileDiffId, message } = req.body;
+
+    const { approval, timestamp, userId, ...profileDiff } = await profileDiffsQuery.fetchProfileDiff(profileDiffId);
```
Comment (pallabez) [inline controllers/users.js:201]: I believe we should handle error here when we get we get empty data and remove the error handler at line 211
```js
if (profileResponse.notFound) return res.boom.notFound("Profile Diff doesn't exist");
```

File: controllers/users.js [hunk 3]
```diff
@@ -196,18 +196,16 @@ const postUserPicture = async (req, res) => {
...
+    if (!profileDiff) return res.boom.notFound("Profile Diff doesn't exist");
```
Comment (pallabez) [inline controllers/users.js:202]: Does this throw `notFound` error when we pass a random profileDiffId? Was it tested?

Comment (RahulGoyal-tech) [reply controllers/users.js:202]: This function doesn't check for that. It only checks if profileDiff is empty the throws an error.

Comment (pallabez) [reply controllers/users.js:202]: So can we do something like this? 
```js
    const profileDiffData = await profileDiffsQuery.fetchProfileDiff(profileDiffId);
    if (!profileDiffData) return res.boom.notFound("Profile Diff doesn't exist");
    
    const { approval, timestamp, userId, ...profileDiff } = profileDiffData;
```

Comment (RahulGoyal-tech) [reply controllers/users.js:202]: What will be the advantage of this change?

Comment (pallabez) [reply controllers/users.js]: Try passing a random profileDiffId and compare the result.

Comment (RahulGoyal-tech) [reply controllers/users.js]: Got it 👍

**Review by pallabez — APPROVED** [review]
Looks good to me :+1:

**Comment by RahulGoyal-tech** [comment]
@lakshayman when will we be merging this PR.

File: controllers/users.js [hunk 4]
```diff
@@ -196,18 +196,18 @@ const postUserPicture = async (req, res) => {
  */
 const updateUser = async (req, res) => {
   try {
-    const userId = req.params.userId;
-    const { id: profileDiffId, message, ...profileDiff } = req.body;
+    const { id: profileDiffId, message } = req.body;
+
+    const profileDiffData = await profileDiffsQuery.fetchProfileDiff(profileDiffId);
+    if (!profileDiffData) return res.boom.notFound("Profile Diff doesn't exist");
```
Comment (Neha) [inline controllers/users.js:202]: Move hardcoded content to constant file

File: controllers/users.js [hunk 5]
```diff
@@ -196,18 +196,18 @@ const postUserPicture = async (req, res) => {
...
+
+    const { approval, timestamp, userId, ...profileDiff } = profileDiffData;
 
     const user = await userQuery.fetchUser({ userId });
     if (!user.userExists) return res.boom.notFound("User doesn't exist");
```
Comment (Neha) [inline controllers/users.js:207]: Personal view: Do not drop {} it is hard for folks to read. We use bundler and they take care of the optimization of the code.

Rule: Make it easy to read.

**Pull request merged**